packages/opentelemetry-tracing/src/BasicTracerRegistry.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConsoleLogger } from '@opentelemetry/core';
+import * as types from '@opentelemetry/types';
+import { Logger } from '@opentelemetry/types';
+import { SpanProcessor, Tracer } from '.';
+import { DEFAULT_CONFIG } from './config';
+import { MultiSpanProcessor } from './MultiSpanProcessor';
+import { NoopSpanProcessor } from './NoopSpanProcessor';
+import { TracerConfig } from './types';
+
+export class BasicTracerRegistry implements types.TracerRegistry {
+  private readonly _registeredSpanProcessors: SpanProcessor[] = [];
+  private readonly _tracers: Map<string, Tracer> = new Map();
+
+  activeSpanProcessor = new NoopSpanProcessor();
+  readonly logger: Logger;
+
+  constructor(private _config: TracerConfig = DEFAULT_CONFIG) {
+    this.logger = _config.logger || new ConsoleLogger(_config.logLevel);
+  }
+
+  getTracer(name = '', version = '*', config?: TracerConfig): Tracer {
+    const key = `${name}@${version}`;
+    if (!this._tracers.has(key)) {
+      this._tracers.set(key, new Tracer(config || this._config, this));
```

Comment:
What if I use the same `name` and `version` but different `config`? Looks like it will ignore the new `config` and return a `tracer` object with previous config, is this expected behavior?

Reply:
It is. I can either document it or we can discuss how to change it. one issue with allowing tracer to be overwritten is then the tracer registry loses the reference to it.

Reply:
Do we have use-cases for this to happen?

Reply:
This is just a hypothetical scenario, imagine I have an application running which is talking with two microservices running one w/ `B3Format` propagator and the other w/ `HTTPTraceContext`. In this case, it is possible to have the same tracer name(=opentelemetry.http) and version (=*) but different config. WDYT? Correct me if I am missing anything.

Reply:
In that case, the `http` plugin would need to get a different tracer depending on which service it was calling. The `http` plugin is just going to call `getTracer('http', '12.0.0')` with no config options. The tracer registry is responsible for making sure that tracer is configured properly. The only way i can see to make that feasible would be to allow `B3` and `TraceContext` to run in parallel on the same tracer (same scope manager, multiple propagators).